src/lib/api/base.ts

```diff
@@ -34,24 +34,43 @@ const api = (linkKey: string) => {
     update: (object: any, url?: string) => {
       return fetch(url || object.links.self, {
         method: 'PATCH',
-        body: JSON.stringify(trimWhitespace(object)),
+        body: JSON.stringify(object),
         headers: { 'Content-Type': 'application/json' },
       })
         .then(checkStatus)
         .then(parseJSON)
         .then(data => data.data)
     },
     create: (object: any, budget: Budget, url?: string) => {
+      // Are we creating a single resource?
+      let single = false
+
+      // If the object is not an array yet, make it one
+      if (!Array.isArray(object)) {
+        single = true
+        object = [object]
+      }
+
+      object = object.map((entry: any) => {
+        return { ...entry, budgetId: budget.id }
+      })
```

Comment:
```suggestion
      // Are we creating a single resource?
      let single = !Array.isArray(object)

      // If the object is not an array yet, make it one
      object = [].concat(object).map((entry: any) => {
        return { ...entry, budgetId: budget.id }
      })
```
This feels a bit more elegant. Not sure if it makes it to hard to read though, what do you think?

Reply:
Looks good to me, I found the initial code I wrote clunky anyway, so this is definitely an improvement.